### DISCUSSION TITLE
How to correctly implement HTTPS communication from a browser extension?

### DISCUSSION THREAD

**Discussion by bytewife** [coddingtonbear/obsidian-local-rest-api#32] [Q&A] [answered]
Hi! I'm wondering if you can help me (an HTTPS/browser novice) get un-stuck with an issue I've been having while using this plugin to communicate with a browser extension 😅. Decided to post this here so that others may read this as well.

I've been building a [browser extension](https://github.com/ivyraine/obsidian-browser2note) that reads Obsidian files. Sending Obsidian GET requests involving auth work in insecure mode, but only in a `web-ext`-opened browser and not in browsers opened normally. I'm thinking the issue is related to CORS, so I'm trying to add HTTPS. However, I am still unable to access endpoints that require auth, such as `https://127.0.0.1:27124/vault/` or opening a file. I'm using [this function](https://github.com/coddingtonbear/obsidian-web/blob/master/src/utils.ts#:~:text=export%20async%20function%20obsidianRequest) to send the GET request. How would you go about reading files from a web browser?

Note: I added this plugin's certificate as a browser CA, but the issue remained.

**Comment by coddingtonbear** [comment]
It's a little bit difficult to say, but one thing you could check is that you've declared which URLs you'd like to interact with in your manifest (e.g. https://github.com/coddingtonbear/obsidian-web/blob/master/public/manifest.json#L38).  You may need to remove/re-add your browser extension after changing that, too.

**Comment by bytewife** [comment, marked as answer]
Ah my issue turned out to be unrelated to the HTTPS connection. If anyone wants another code reference you can check out the browser extension listed above!
